File: backend-ts/src/services/chatService.ts
import { supabaseAdmin } from "../supabase.js";
import type { Chat, TablesInsert } from "../types/supabase.js";

export async function listChats(caseId: string): Promise<Chat[]> {
  const { data, error } = await supabaseAdmin
    .from("chats")
    .select("*")
    .eq("case_id", caseId)
    .order("created_at", { ascending: false });
  if (error) {
    throw error;
  }
  return data as Chat[];
}

export async function createChat(payload: TablesInsert["chats"]): Promise<Chat> {
  const { data, error } = await supabaseAdmin.from("chats").insert(payload).select().single();
  if (error) {
    throw error;
  }
  return data as Chat;
}

export async function getChat(chatId: string): Promise<Chat | null> {
  const { data, error } = await supabaseAdmin
    .from("chats")
    .select("*")
    .eq("id", chatId)
    .maybeSingle();
  if (error) {
    throw error;
  }
  return (data as Chat) ?? null;
}

export async function deleteChat(chatId: string): Promise<void> {
  const { error } = await supabaseAdmin.from("chats").delete().eq("id", chatId);
  if (error) {
    throw error;
  }
}

File: backend/app/services/chat_service.py
from __future__ import annotations

from ..models import Chat
from .supabase_service import SupabaseService


class ChatService:
    def __init__(self, supabase: SupabaseService | None = None):
        self.supabase = supabase or SupabaseService()

    def list_chats(self, case_id: str) -> list[Chat]:
        records = self.supabase.select("chats", filters={"case_id": case_id}, order=("created_at", False))
        return [Chat(**record) for record in records]

    def create_chat(self, case_id: str, title: str) -> Chat:
        record = self.supabase.insert("chats", {"case_id": case_id, "title": title})[0]
        return Chat(**record)

    def get_chat(self, chat_id: str) -> Chat | None:
        record = self.supabase.select_by_id("chats", chat_id)
        return Chat(**record) if record else None

    def delete_chat(self, chat_id: str) -> None:
        self.supabase.delete("chats", chat_id)

